src/System.Text.Json/src/System/Text/Json/Node/JsonString.cs

```diff
@@ -208,5 +209,69 @@ public string Value
         ///   Returns <see cref="JsonValueKind.String"/>
         /// </summary>
         public override JsonValueKind ValueKind { get => JsonValueKind.String; }
+
+        /// <summary>
+        ///   Converts the text value of this instance, which should encode binary data as base-64 digits, to an equivalent 8-bit unsigned <see cref="byte"/> array.
+        ///   The return value indicates wether the conversion succeeded.
+        /// </summary>
+        /// <param name="value">
+        ///   When this method returns, contains the <see cref="byte"/> array equivalent of the text contained in this instance,
+        ///   if the conversion succeeded.
+        /// </param>
+        /// <returns>
+        ///   <see langword="true"/> if text was converted successfully; othwerwise returns <see langword="false"/>.
+        /// </returns>
+        internal bool TryGetBytesFromBase64(out byte[] value)
+        {
+            if (_value.Length < 4)
+            {
+                value = default;
+                return false;
+            }
+
+#if BUILDING_INBOX_LIBRARY
+            // we decode string -> byte, so the resulting length will
+            // be /4 * 3 - padding. To be on the safe side, keep padding and slice later
+            int bufferSize = _value.Length / 4 * 3;
+
+            byte[] arrayToReturnToPool = null;
+            try
+            {
+                Span<byte> buffer = bufferSize <= JsonConstants.StackallocThreshold
+                    ? stackalloc byte[bufferSize]
+                    : arrayToReturnToPool = ArrayPool<byte>.Shared.Rent(bufferSize);
+
+                if (Convert.TryFromBase64String(_value, buffer, out int bytesWritten))
```

Comment:
Will `TryFromBase64String` ever throw? I don't think so (other than null input, which we already know to be true). Maybe we don't need a try/finally.

1) Add `Debug.Assert(_value != null)`
2) Consider removing the try/finally? I don't feel strongly about this, so it's fine to keep it.

Reply:
I'll look into it. It would be nice to remove the try-finally blocks.

Reply:
The try-finally blocks are now removed. Nothing should throw exceptions, with the assumption that `bytesWritten` can't be larger than the length of the buffer. I'll leave this unresolved until someone else can confirm that they find no potential exceptions.